core/common/src/main/scala/org/typelevel/otel4s/TextMapPropagator.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+trait TextMapPropagator[F[_]] extends TextMapInjector[F] with TextMapExtractor
+
+trait TextMapInjector[F[_]] {
+  def inject[A](context: Context, carrier: A)(implicit
+      setter: TextMapSetter[F, A]
+  ): F[A]
+}
+
+trait TextMapExtractor {
+  def extract[A](context: Context, carrier: A)(implicit
+      getter: TextMapGetter[A]
+  ): Context
+}
+
+trait TextMapCarrier[F[_], A] extends TextMapGetter[A] with TextMapSetter[F, A]
+
+trait TextMapGetter[A] {
+  def keys(carrier: A): List[String]
+  def get(carrier: A, key: String): Option[A]
```

Comment:
@mccoy and I had a workplace accident this week relating to this being case sensitive, but it's not clear that the spec supports case insensitivity for all transports.